src/query.rs

```diff
@@ -1794,6 +1783,40 @@ fn release_borrow<Q: Query>(archetypes: &[Archetype]) {
     }
 }
 
+struct ArchetypeIter<Q: Query> {
+    archetypes: core::ops::Range<usize>,
+    _marker: PhantomData<Q>,
+}
+
+impl<Q: Query> ArchetypeIter<Q> {
+    fn new(world: &World) -> Self {
+        Self {
+            archetypes: 0..world.archetypes().len(),
+            _marker: PhantomData,
+        }
+    }
+
+    fn next(&mut self, world: &World) -> Option<ChunkIter<Q>> {
```

Comment:
It isn't API, but I wonder if this needs to be `unsafe` as passing the wrong `world` could mean the `get_unchecked` is out of bounds?

Reply:
Good catch, thanks.